js/src/setup-ads/ads-stepper/setup-accounts/google-ads-account-section/create-account-button.js

```diff
@@ -10,11 +10,23 @@ import { useState } from '@wordpress/element';
 import TermsModal from './terms-modal';
 import AppButton from '.~/components/app-button';
 
+/**
+ * Renders a Google Ads account creaton button.
+ * When clicking on the button, it will pop up a modal first to ask for terms agreement.
+ *
+ * @param {Object} props React props.
+ * @param {() => boolean|void} [props.onBeforeAskTerms] Called before showing the terms agreement modal.
+ *     Return `true` to interrupt the ask for terms agreement and the subsequent process.
+ * @param {Function} [props.onCreateAccount] Called after the user accept the terms agreement.
+ */
 const CreateAccountButton = ( props ) => {
-	const { onCreateAccount, ...rest } = props;
+	const { onBeforeAskTerms, onCreateAccount, ...rest } = props;
 	const [ isOpen, setOpen ] = useState( false );
 
 	const handleButtonClick = () => {
+		if ( onBeforeAskTerms && onBeforeAskTerms() === false ) {
```

Comment:
💅  I always feel this looks easier on the eyes (to cut down the check on `onBeforeAskTerms` at line 27 ) 😄 

```suggestion
	const { onBeforeAskTerms = () => {}, onCreateAccount, ...rest } = props;
	const [ isOpen, setOpen ] = useState( false );

	const handleButtonClick = () => {
		if ( onBeforeAskTerms() === false ) {
```

Reply:
I added a `noop` function as its default value by https://github.com/woocommerce/google-listings-and-ads/pull/1069/commits/fa9f0a8edee02b786800935115029c9953654468. If want to omit the check, giving a `noop` function could have more semantics and keep the same reference.